src/meta/meta_table.c

```diff
@@ -104,6 +104,8 @@ __wt_metadata_cursor(WT_SESSION_IMPL *session, WT_CURSOR **cursorp)
      * one.
      */
     cursor = NULL;
+    if (cursorp != NULL)
```

Comment:
I think the real solution is to initialize the pointer to `NULL` before calling it in your own code.
So:
```
WT_CURSOR *metadata_cursor;
metadata_cursor = NULL;
__wt_metadata_cursor(session, &metadata_cursor);
```

Reply:
I will add this, but keep this in as Keith suggested for increased safety :)

Reply:
We shouldn't do it in both places. If Alex found a place where I got it wrong, that's cool, but no, not in both places. (I feel strongly.) :)

Reply:
Fair enough! Thanks Keith :)

Reply:
@andym03 Yeah there's no need to do it in both places.

My thinking was is that the function has already indicated failure by returning non-zero. If you think about it, the only reason we want to initialize to zero is for convenience so that the caller can blindly call the close function without checking `ret` so I'd consider it the caller's job to do that for itself.

I can understand you and @keithbostic's point of view though. No worries.